### ISSUE TITLE
Impossible to update fan speed

### ISSUE THREAD

**Issue by oliroche** [mduran80/pymadoka#18]
Amazing piece of code, still trying to understand the issue, however:

When I try to update the fan-speed from MQTT, I get "ERROR:root:Could not update operation mode: <FanSpeedEnum.LOW: 1>". It specifically seems to break when I enter 'MID'. pymadoka works perfect on the command line.

Here is what I get:

/usr/local/lib/python3.10/dist-packages/pymadoka/connection.py:173: FutureWarning: This method will be removed future version, pass the callback to the BleakClient constructor instead.
  self.client.set_disconnected_callback(self.on_disconnect)
ERROR:pymadoka.mqtt:Operation cancelled : 
ERROR:root:Could not update operation mode: 'NoneType' object has no attribute 'cooling_fan_speed'
ERROR:root:Could not update operation mode: <FanSpeedEnum.MID: 3>

Any clue?
Cheers,
Olivier.

**Comment by mduran80** [comment]
It seems the controller does not have a fan status for the cooling mode and
the code assumes it always exists.

https://github.com/mduran80/pymadoka/blob/4486d9f3d2891d94ea23980d698f2326c691b6aa/pymadoka/mqtt.py#L99

I don’t have a mqtt setup myself. Can you check if the cooling_fan_speed is
being reported in the status or fan topic? From what I see in the code, it
should. But it seems the fan status is not being correctly initialized

**Comment by oliroche** [comment]
So, if I update target temperature, MQTT get updated no issue.

If I ssh: pymadoka -a F0:B3:1E:D4:EC:E2 set-fan-speed HIGH HIGH (with pymadoka-mqtt running)

I correctly get: {"cooling_fan_speed": « HIGH", "heating_fan_speed": « HIGH"}

I get in MQTT explore, I see no update. If I ctrl C on the pymadoka-mqtt & relaunch, then MQTT gets updated. From there, all the update through SSH show in the MQTT. However, as soon as I set: 

madoka/F0_B3_1E_D4_EC_E2/fan_speed/set to LOW for example, I get: ERROR:root:Could not update operation mode: <FanSpeedEnum.MID: 3> in pymadoka-mqtt log, and then no more updates to the fan topic.

**Comment by mduran80** [comment]
https://github.com/mduran80/pymadoka/blob/4486d9f3d2891d94ea23980d698f2326c691b6aa/pymadoka/mqtt.py#L213

I think that uppercase MEDIUM can be the culprit. Try to replace it with
MID

**Comment by oliroche** [comment]
Where should I replace? I use « MID » when I try to change the topic. You mean I shall change in MQTT.py?

**Comment by mduran80** [comment]
Yes, update the file mqtt.py and replace MEDIUM with MID and check if that
works. If so, i will pish a new version of pymadoka with the fix.

**Comment by oliroche** [comment]
Tested with no luck, will try again in a few days.

**Comment by mduran80** [comment]
I realized it is not easy to test because HA downloads pymadoka package and
it will never contain your code. I will push a version this weekend and
will let you know so you can test (bumping the pymadoka dependency version)
